keras_cv/ops/box_matcher_test.py

```diff
@@ -0,0 +1,127 @@
+        bg_thresh_lo = 0.0
+
+        with self.assertRaises(ValueError):
+            _ = ArgmaxBoxMatcher(
+                thresholds=[bg_thresh_hi, bg_thresh_lo, fg_threshold],
+                match_values=[-3, -2, -1],
+            )
+
+    def test_box_matcher_unbatched(self):
+        sim_matrix = tf.constant([[0.04, 0, 0, 0], [0, 0, 1.0, 0]], dtype=tf.float32)
+
+        fg_threshold = 0.5
+        bg_thresh_hi = 0.2
+        bg_thresh_lo = 0.0
+
+        matcher = ArgmaxBoxMatcher(
+            thresholds=[bg_thresh_lo, bg_thresh_hi, fg_threshold],
+            match_values=[-3, -2, -1, 1],
+        )
+        match_indices, matched_values = matcher(sim_matrix)
+        positive_matches = tf.greater_equal(matched_values, 0)
+        negative_matches = tf.equal(matched_values, -2)
+
+        self.assertAllEqual(positive_matches.numpy(), [False, True])
+        self.assertAllEqual(negative_matches.numpy(), [True, False])
+        self.assertAllEqual(match_indices.numpy(), [0, 2])
+        self.assertAllEqual(matched_values.numpy(), [-2, 1])
+
+    def test_box_matcher_batched(self):
+        sim_matrix = tf.constant([[[0.04, 0, 0, 0], [0, 0, 1.0, 0]]], dtype=tf.float32)
+
+        fg_threshold = 0.5
+        bg_thresh_hi = 0.2
+        bg_thresh_lo = 0.0
+
+        matcher = ArgmaxBoxMatcher(
+            thresholds=[bg_thresh_lo, bg_thresh_hi, fg_threshold],
+            match_values=[-3, -2, -1, 1],
+        )
+        match_indices, matched_values = matcher(sim_matrix)
+        positive_matches = tf.greater_equal(matched_values, 0)
+        negative_matches = tf.equal(matched_values, -2)
+
+        self.assertAllEqual(positive_matches.numpy(), [[False, True]])
+        self.assertAllEqual(negative_matches.numpy(), [[True, False]])
+        self.assertAllEqual(match_indices.numpy(), [[0, 2]])
+        self.assertAllEqual(matched_values.numpy(), [[-2, 1]])
+
+    def test_box_matcher_force_match(self):
+        sim_matrix = tf.constant(
+            [[0, 0.04, 0, 0.1], [0, 0, 1.0, 0], [0.1, 0, 0, 0], [0, 0, 0, 0.6]],
+            dtype=tf.float32,
+        )
+
+        fg_threshold = 0.5
+        bg_thresh_hi = 0.2
+        bg_thresh_lo = 0.0
+
+        matcher = ArgmaxBoxMatcher(
+            thresholds=[bg_thresh_lo, bg_thresh_hi, fg_threshold],
+            match_values=[-3, -2, -1, 1],
+            force_match_for_each_col=True,
+        )
+        match_indices, matched_values = matcher(sim_matrix)
+        positive_matches = tf.greater_equal(matched_values, 0)
+        negative_matches = tf.equal(matched_values, -2)
+
+        self.assertAllEqual(positive_matches.numpy(), [True, True, True, True])
+        self.assertAllEqual(negative_matches.numpy(), [False, False, False, False])
+        # the first anchor cannot be matched to 4th gt box given that is matched to
+        # the last anchor.
+        self.assertAllEqual(match_indices.numpy(), [1, 2, 0, 3])
+        self.assertAllEqual(matched_values.numpy(), [1, 1, 1, 1])
```

Comment:
since we pick 0.04 for first row and 0.1 for the third row, shouldn't the match_values be [-2, 1, -2, 1] (based on the bucket)?

Reply:
No, this is when each col needs a match, so it is all 1